codegen-core/src/test/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/BuilderGeneratorTest.kt

```diff
@@ -30,29 +34,34 @@ internal class BuilderGeneratorTest {
     @Test
     fun `generate builders`() {
         val provider = testSymbolProvider(model)
-        val writer = RustWriter.forModule("model")
-        writer.rust("##![allow(deprecated)]")
-        val innerGenerator = StructureGenerator(model, provider, writer, inner)
-        val generator = StructureGenerator(model, provider, writer, struct)
-        val builderGenerator = BuilderGenerator(model, provider, struct)
-        generator.render()
-        innerGenerator.render()
-        builderGenerator.render(writer)
-        writer.implBlock(struct, provider) {
-            builderGenerator.renderConvenienceMethod(this)
+        val project = TestWorkspace.testProject(provider)
+        project.moduleFor(inner) {
+            rust("##![allow(deprecated)]")
```

Comment:
Thanks to my attributes refactor you can now write:
```suggestion
            AllowDeprecated.render(this)
```

Is it an improvement? 🤷‍♀️ 
you decide